src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -797,6 +797,70 @@ void ClearSecretData(uint8_t * buf, uint32_t len)
     memset(buf, 0, len);
 }
 
+CHIP_ERROR GenP256Keypair(P256PublicKey * pubkey, P256PrivateKey * privkey)
+{
+    ERR_clear_error();
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    int result       = 0;
+    int nid          = NID_undef;
+    EC_KEY * ec_key  = nullptr;
+    EC_GROUP * group = nullptr;
+
+    VerifyOrExit(pubkey != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
if this function accepted references instead of pointers, you wouldn't have to check for NULL

Reply:
That's a good suggestion. I've updated the code.